lib/inspec/plugins/fetcher.rb

```diff
@@ -11,96 +11,28 @@ def self.plugin_registry
         Inspec::Fetcher
       end
 
-      # Provide a list of files that are available to this fetcher.
-      #
-      # @return [Array[String]] A list of filenames
-      def files
-        fail "Fetcher #{self} does not implement `files()`. This is required."
-      end
-
-      # Read a file using this fetcher. The name must correspond to a file
-      # available to this fetcher. Use #files to retrieve the list of
-      # files.
-      #
-      # @param [String] _file The filename you are interested in
-      # @return [String] The file's contents
-      def read(_file)
-        fail "Fetcher #{self} does not implement `read(...)`. This is required."
-      end
-
-      def relative_target
-        RelFetcher.new(self)
+      def archive_path
+        fail "Fetcher #{self} does not implement `archive_path()`. This is required."
       end
-    end
-
-    BLACKLIST_FILES = [
-      '/pax_global_header',
-      'pax_global_header',
-    ].freeze
-
-    class RelFetcher < Fetcher
-      attr_reader :files
-      attr_reader :prefix
 
-      def initialize(fetcher)
-        @parent = fetcher
-        @prefix = get_prefix(fetcher.files)
-        @files = fetcher.files.find_all { |x| x.start_with? prefix }
-                        .map { |x| x[prefix.length..-1] }
+      def fetch(_path)
+        fail "Fetcher #{self} does not implement `fetch()`. This is required."
       end
 
-      def abs_path(file)
-        return nil if file.nil?
-        prefix + file
+      def resolved_source
+        fail "Fetcher #{self} does not implement `resolved_source()`. This is required for terminal fetchers."
       end
 
-      def read(file)
-        @parent.read(abs_path(file))
+      def cacheable?
```

Comment:
This is unused,remove.  Wether or not a fetcher adds items to the cache is determined by what it does with the argument to `fetch/1`